### PR TITLE
Add explicit System.Memory dependency to Hosts

### PR DESCRIPTION
**Pull request by JoeRobich**
Resolves https://github.com/OmniSharp/omnisharp-vscode/issues/5352

Similar to https://github.com/dotnet/roslyn/pull/63545. O# needs to have the newer version of System.Memory available when running against newer releases of VS Build Tools or .NET SDKs.

Also updates SDKs to a version that would have issues loading.

### CODE REVIEW

**Comment by nickspoons** [comment]
From the error logs it looks like this PR will also fix https://github.com/OmniSharp/omnisharp-roslyn/issues/2442 and https://github.com/OmniSharp/omnisharp-vim/issues/803

File: src/OmniSharp.Http.Driver/app.config [hunk 1]
```diff
@@ -62,6 +62,11 @@
                 <assemblyIdentity name="System.Composition.TypedParts" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
                 <bindingRedirect oldVersion="0.0.0.0-6.0.0.0" newVersion="6.0.0.0"/>
             </dependentAssembly>
+
+            <dependentAssembly>
+                <assemblyIdentity name="System.Memory" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
+                <bindingRedirect oldVersion="0.0.0.0-4.0.1.2" newVersion="4.0.1.2" />
```
Comment (333fred) [inline src/OmniSharp.Http.Driver/app.config:68]: Why doesn't this version match up with the dependency version?

Comment (JoeRobich) [reply src/OmniSharp.Http.Driver/app.config:68]: Because the assembly version doesn't match the nuget package version.

Comment (333fred) [reply src/OmniSharp.Http.Driver/app.config:68]: Might be worth a comment.

**Review by 333fred — APPROVED** [review]

**Pull request merged**